js/components/app.js

```diff
@@ -85,11 +87,16 @@ export default class App extends PureComponent {
     });
   }
 
+  getInteractiveState(data) {
+    this.phone.post('interactiveState', this.state.interactiveState);
+  }
+
   handleAuthoredStateChange(state) {
     this.phone.post('authoredState', state);
   }
 
   handleInteractiveStateChange(state) {
+    this.state.interactiveState = state;
```

Comment:
In general we should never change React state like this (mutate). It's should always done using `this.setState` method, so React can re-render component:
```javascript
this.setState({interactiveState: state});
```
However, in this particular case we don't really need to use React state and re-render at all, as new `interactiveState` doesn't really change UI / dom structure. Actually I should have call this state property `initialInteractiveState` rather than `interactiveState`, so it would be cleaner. So, instead we could just say:
```javascript
this._currentInteractiveState = state;
```
and then use it in `getInteractiveState`. But either way will work. Your solution also works, as there's no need to re-render component in this particular case.

It might be a bit confusing - that what happens when you mix UI with domain logic, but I feel it's justified and handy in tiny apps (where libs like Redux could be an overkill).